bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java

```diff
@@ -134,7 +134,7 @@ public Integer getRank() {
             StateDescription dynamicStateDescription = provider.getStateDescription(channel, originalStateDescription,
                     locale);
             if (dynamicStateDescription != null) {
-                if (dynamicStateDescription == originalStateDescription) {
```

Comment:
Can we revert this? There was a reason to use the comparison against the reference here: https://github.com/openhab/openhab-core/pull/1817

Reply:
Good catch! I'll restore it and add a comment so we won't run into this regression again. :blush: